### PR TITLE
Add step to fail fast scenario

### PR DESCRIPTION
**Pull request by ikalinin1**

### CODE REVIEW

File: vividus-plugin-web-app/src/main/java/org/vividus/bdd/steps/ui/web/validation/HighlightingSoftAssert.java [hunk 1]
```diff
@@ -196,8 +197,15 @@ public void verify() throws VerificationError
...
+    public void verify(Pattern pattern) throws VerificationError
+    {
+        descriptiveSoftAssert.verify(pattern);
+    }
+
     private void clearAssertingWebElements()
     {
         webUiContext.clearAssertingWebElements();
     }
+
```
Comment (valfirst) [inline vividus-plugin-web-app/src/main/java/org/vividus/bdd/steps/ui/web/validation/HighlightingSoftAssert.java]: useless

File: vividus/src/main/java/org/vividus/bdd/steps/AssertSteps.java [hunk 2]
```diff
@@ -44,4 +46,17 @@ public void verifyIfAssertionsPassed()
...
     }
+
+    /**
+     * Steps walks the failed assertions and if any of assertion messages matches regex
+     * all the failed assertions verified;
+     * Step could be useful when you have to fail-fast scenario in case of some assertion;
+     * 
+     * @param assertionPatter to match assertions
+     */
+    @Then("I verify matching '$assertionPattern' assertions")
```
Comment (valfirst) [inline vividus/src/main/java/org/vividus/bdd/steps/AssertSteps.java]: `@Then("I verify assertions matching '$assertionPattern'")` ?

File: vividus-tests/src/main/resources/story/integration/AssertStepsTests.story [hunk 3]
```diff
@@ -0,0 +1,11 @@
+Narrative: Story skipped for now. Because to verify the feature we have to use known-issues, but with known issue exit code = 1
+For now we cannot override this beahvior. Details in the PR https://github.com/gradle/gradle/pull/10763
+
+Scenario: Set-Up
+Given I am on a page with the URL 'https://google.com'
+
+Scenario: Verify step Then I verify matching '$assertionPattern' assertions
```
Comment (valfirst) [inline vividus-tests/src/main/resources/story/integration/AssertStepsTests.story]: Scenario: Verify step Then I verify assertions matching '$assertionPattern'

File: vividus-tests/src/main/resources/known-issues.json [hunk 4]
```diff
@@ -1,2 +1,8 @@
 {
+  "VVD-1": {
+    "type": "Internal",
+    "assertionPattern": ".*Doctor Who.*",
+    "scenarioPattern": "Verify step Then I verify matching '.+' assertions",
```
Comment (valfirst) [inline vividus-tests/src/main/resources/known-issues.json]: "scenarioPattern": "Verify step Then I verify assertions matching '.+'",

File: vividus-soft-assert/src/main/java/org/vividus/softassert/SoftAssert.java [hunk 5]
```diff
@@ -288,6 +289,19 @@ public void verify() throws VerificationError
         }
     }
 
+    @Override
+    public void verify(Pattern pattern) throws VerificationError
+    {
+        if (getAssertionCollection().getAssertionErrors().stream()
```
Comment (uarlouski) [inline vividus-soft-assert/src/main/java/org/vividus/softassert/SoftAssert.java]: put here NOT op and move verify inside if condition? just to reduce number of code lines by 1

File: vividus-soft-assert/src/main/java/org/vividus/softassert/SoftAssert.java [hunk 6]
```diff
@@ -288,6 +289,19 @@ public void verify() throws VerificationError
...
+                                                         .map(SoftAssertionError::getError)
+                                                         .map(AssertionError::getMessage)
+                                                         .noneMatch(m -> pattern.matcher(m).matches()))
```
Comment (uarlouski) [inline vividus-soft-assert/src/main/java/org/vividus/softassert/SoftAssert.java]: pattern.asMatchPredicate?

File: vividus-soft-assert/src/main/java/org/vividus/softassert/SoftAssert.java (hunk 5 above)
Comment (ikalinin1) [reply vividus-soft-assert/src/main/java/org/vividus/softassert/SoftAssert.java]: I will not, I like the readability of the current version

**Pull request merged**